### PR TITLE
Added an option for exclude presigned url not_sent request usage in api_request counts

### PR DESCRIPTION
**Pull request by cjyclaire**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Addressing #1866

### CODE REVIEW

**Comment by awood45** [comment]
We should also carefully call out in release notes how tests could break when updating to this - if customers are relying on this to be collected for presigned requests, they'll need to adapt.

**Comment by awood45** [comment]
On second thought, hold on merging this. I don't want to cause surprise test breakage if there is another way to avoid this.

File: gems/aws-sdk-s3/spec/presigner_spec.rb [hunk 1]
```diff
@@ -43,6 +43,20 @@ module S3
...
+          presigner = Presigner.new(client: stub_client)
+          presigner.presigned_url(:get_object, bucket: 'bkt', key: 'k')
+          expect(stub_client.api_requests.size).to eq(1)
+        end
+
+        it 'can be excluded from being tracked as an api request' do
+          stub_client = Aws::S3::Client.new(stub_responses: true, exclude_presign: true)
+          presigner = Presigner.new(client: stub_client)
+          presigner.presigned_url(:get_object, bucket: 'bkt', key: 'k')
+          expect(stub_client.api_requests.size).to be_zero
```
Comment (awood45) [inline gems/aws-sdk-s3/spec/presigner_spec.rb]: I actually imagined this as an option you could pass in to the `api_requests` method itself. Should be easy to replace the attr_reader with a method that takes an optional param to do this. That way, you don't need to decide at client construction time.

**Comment by cjyclaire** [comment]
Also addressed doc issue mentioned in #1868

**Pull request merged**